File: tools/census-harvest/tests/fixtures/ts-mini/src/engine.test.ts
import { compact } from "./engine";
import { Store } from "./store";

function compactHelper(store: Store): number {
  return store.stats().objects;
}

export async function testCompact(): Promise<void> {
  const s = new Store({ maxSize: 8, region: "eu" });
  const n = await compact(s, "k");
  if (n !== compactHelper(s)) {
    throw new Error("mismatch");
  }
}

File: plugins/igr/tools/census-harvest/tests/fixtures/go-mini/engine/engine_test.go
package engine

import (
	"testing"

	"example.com/gomini/store"
)

func compactHelper(s *store.Store) uint64 {
	return s.Stats().Objects
}

func TestCompact(t *testing.T) {
	s := store.New(store.Config{MaxSize: 8, Region: "eu"})
	n, err := Compact(s, "k")
	if err != nil {
		t.Fatal(err)
	}
	if n != compactHelper(s) {
		t.Fatalf("mismatch")
	}
}
